resolver/resolver.go

```diff
@@ -281,9 +281,9 @@ func (t Target) Endpoint() string {
 	return strings.TrimPrefix(endpoint, "/")
 }
 
-// String returns a string representation of Target.
+// String returns a canonical string representation of Target.
```

Comment:
Super nit: "_the_ canonical", please... there is only one canonical version of something.